pyigm/clustering/xcorr_utils.py

```diff
@@ -222,3 +225,136 @@ def clean_matrix(W,n=-99):
     W = np.where(np.isnan(W),n,W)
     W = np.where(np.isinf(W),n,W)
     return W
+
+
+def random_gal(galreal, Nrand, Nmin=20, DZ=0.01, smooth_scale=10.):
+    """
+    Prefered random galaxy generator. For a given galaxy with a
+    given magnitude (and other properties), it calculates the redshift
+    sensitivity function from galaxies in a magnitude band around the
+    selected one (i.e., including slightly brighter and fainter
+    galaxies), and places Nrand new galaxies at a random redshift given
+    by a smoothed version of the observed sensitivity function. For
+    extremely bright or faint galaxies (rare) the sensitivity function
+    is calculated from at least Nmin (=20) galaxies (i.e. the magnitude
+    band is increased).
+
+    Parameters
+    ----------
+    galreal
+    Nrand
+    Nmin : int, optional
+    DZ : float, optional
+      delta z for the histogram for getting the
+    smooth_scale : float, optional
+       smoothing scale for the histogram (in number of bins, so depends on DZ)
+
+    Returns
+    -------
+    galrand : np rec array
+      Copy of galreal, Nrand times
+    """
+    from pyigm.clustering.randist import RanDist
+    from scipy.interpolate import CubicSpline
+    from scipy.ndimage import gaussian_filter as gf
+
+    debug = 0
+    Ckms = 299792.458
```

Comment:
get C from astropy constants?

Reply:
I thought about it, but that seems overkill.

Just made it a global, though with a comment.